src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector512_1.cs

```diff
@@ -0,0 +1,391 @@
+        /// <param name="vector">The vector to negate.</param>
+        /// <returns>A vector whose elements are the unary negation of the corresponding elements in <paramref name="vector" />.</returns>
+        /// <exception cref="NotSupportedException">The type of the vector (<typeparamref name="T" />) is not supported.</exception>
+        [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector512<T> operator -(Vector512<T> vector)
+        {
+            return Vector512.Create(
+                -vector._lower,
+                -vector._upper
+            );
+        }
+
+        /// <summary>Returns a given vector unchanged.</summary>
+        /// <param name="value">The vector.</param>
+        /// <returns><paramref name="value" /></returns>
+        /// <exception cref="NotSupportedException">The type of the vector (<typeparamref name="T" />) is not supported.</exception>
+        [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector512<T> operator +(Vector512<T> value)
+        {
+            ThrowHelper.ThrowForUnsupportedIntrinsicsVector512BaseType<T>();
+            return value;
+        }
+
+        /// <summary>Determines whether the specified object is equal to the current instance.</summary>
+        /// <param name="obj">The object to compare with the current instance.</param>
+        /// <returns><c>true</c> if <paramref name="obj" /> is a <see cref="Vector512{T}" /> and is equal to the current instance; otherwise, <c>false</c>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public override bool Equals([NotNullWhen(true)] object? obj) => (obj is Vector512<T> other) && Equals(other);
+
+        /// <summary>Determines whether the specified <see cref="Vector512{T}" /> is equal to the current instance.</summary>
+        /// <param name="other">The <see cref="Vector512{T}" /> to compare with the current instance.</param>
+        /// <returns><c>true</c> if <paramref name="other" /> is equal to the current instance; otherwise, <c>false</c>.</returns>
+        /// <exception cref="NotSupportedException">The type of the vector (<typeparamref name="T" />) is not supported.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool Equals(Vector512<T> other)
+        {
+            // This function needs to account for floating-point equality around NaN
+            // and so must behave equivalently to the underlying float/double.Equals
+
+            if (Vector512.IsHardwareAccelerated)
+            {
+                if ((typeof(T) == typeof(double)) || (typeof(T) == typeof(float)))
+                {
+                    Vector512<T> result = Vector512.Equals(this, other) | ~(Vector512.Equals(this, this) | Vector512.Equals(other, other));
+                    return result.AsInt32() == Vector512<int>.AllBitsSet;
+                }
```

Comment:
Can you elaborate on this logic?

Reply:
Same logic as already in `Vector64/128/256`.

We say that (`this` is the same as `other`) or ((`this` is `NaN`) and (`other` is not `NaN`)).

Basically, `Equals(this, other)` returns (per element) `AllBitsSet` if the two values are equal and `Zero` if they are not equal or either is `NaN`.

`Equals(this, this)` and `Equals(other, other)` returns (per element) `AllBitsSet` if the element is not NaN and `Zero` otherwise.

We negate the combination of `Equals(this, this) | Equals(other, other)` which gives us (per element) `AllBitsSet` if both `this` and `other` are `NaN` and `Zero` if either input was `not NaN`

When that is combined with the original `Equals(this, other)` we get `AllBitsSet` if the two values are equal or both `NaN` and `Zero` otherwise (they are not equal or either is `not NaN`).